File: sum-of-root-to-leaf-binary-numbers/sum-of-root-to-leaf-binary-numbers.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
     public int sumRootToLeaf(TreeNode root) {
        List<String> list = new ArrayList<>();
        helper(list, root, new StringBuilder().append(Integer.toString(root.val)));
        int res = 0;
        for (String ele: list) {
            res += Integer.parseInt(ele, 2);
        }
        return res;
    }
    private void helper(List<String> list, TreeNode root, StringBuilder sb) {
        if (root == null) return;
        
        if (root.left == null && root.right == null) {
            list.add(sb.toString());
            return;
        }
        
        if (root.left != null) {
            sb.append(Integer.toString(root.left.val));
            helper(list, root.left, sb);
            sb.deleteCharAt(sb.length() - 1);
        }
        if (root.right != null) {
            sb.append(Integer.toString(root.right.val));
            helper(list, root.right, sb);
            sb.deleteCharAt(sb.length() - 1);
        }
    }
}

File: 1079-sum-of-root-to-leaf-binary-numbers/sum-of-root-to-leaf-binary-numbers.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var sumRootToLeaf = function(root) {
    let res = 0;
    // DFS function
    var traverse = (curNode, value = root.val) => {
        // leaf node
        if(curNode.left === null && curNode.right === null){
            res += value;
            return;
        }
        // traverse left child
        if(curNode.left !== null){
            traverse(curNode.left, value * 2 + curNode.left.val);
        }
        // traverse right child
        if(curNode.right !== null){
            traverse(curNode.right, value * 2 + curNode.right.val);
        }
    };
    traverse(root);
    return res;
};